server/controller/chore.js

```diff
@@ -196,3 +220,55 @@ export const deleteChore = async (req, res) => {
     res.status(500).json({ message: "Internal Server Error" });
   }
 }; //deleteChore
+
+/*
+ ** This function checks off a chore by updating its nextOccurrence and checkedOff params
+ ** @param {Object} req - The request object
+ ** @param {Object} res - The response object
+ */
+ export const checkOffChore = async (req, res) => {
+  try {
+    const id = req.params.id;
+
+    const chore = await Chore.findById({ _id: id });
+
+    if (chore === null) {
+      return res
+        .status(404)
+        .json({ message: `Chore with id ${id} was not found` });
+    }
+
+    // mapping interval into days
+    var intervalToDays;
+    if (chore.frequency.interval == "days") {
+        intervalToDays = 1;
+    } else if (chore.frequency.interval == "weeks") {
+        intervalToDays = 7;
+    } else if (chore.frequency.interval == "months") {
+        intervalToDays = 30; // just defaulting to 30
+    } else {
+        intervalToDays = 365; // interval is in years
+    }
```

Comment:
I didn't even see this when I made my other comment. Case in point, this should be a function :P